### ISSUE TITLE
Dispatch not being set by connect for exported component but the actions are

### ISSUE THREAD

**Issue by Prudwell** [reactjs/react-redux#489]
The closest thing I could find was #163 but I don't think mine is the same.

So I have a connected container that loads fine when it is called by my root container using react-native-router-flux

TodoContainer loads fine in this setup when an action tells react-native-router-flux to go there

```
//root.js

export default class Root extends Component {
  render() {
    return (
      <Provider store={this.props.store}>
        <RouterWithRedux>
            <Scene key="initializeLocalUser" hideNavBar component={InitializeLocalUser} title="Initialize" sharedResources={this.props.sharedResources}/>
            <Scene key="todos" component={TodoContainer} title="Todos" sharedResources={this.props.sharedResources} />
        </RouterWithRedux>
      </Provider>
    );
  }
}
```
 Here's a pared down version of the TodoContainer.js

```
import React, { Component } from 'react'
import { View, Text} from 'react-native'
import { appStyle } from '../styles/styles';
import { getLocalUserFromDatabase, fetchTodosIfNeeded, deleteTodo } from '../../actions/nativeActions.js'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'


class TodoContainer extends Component {
    constructor(props) {
        super(props)
    }
    componentDidMount() {
        const { dispatch, view } = this.props
        dispatch(fetchTodosIfNeeded('todoList', this.props.sharedResources.database))
    }
    render() {
        return(
        <View style={appStyle.container}>
            <Text>HI HI</Text>
        </View>
        )
    }
}

const dispatchToProps = (dispatch) => {
    return bindActionCreators({
        fetchTodosIfNeeded,
        deleteTodo
    }, dispatch)
}

export default connect(state => state, dispatchToProps)(TodoContainer)
```

However when I try to load the TodoContainer via a Scene that loads a new Dashboard component (not connected) I run into _Possible Unhandled Promise Rejection (id: 0): dispatch is not a function_

```
//new root.js

export default class Root extends Component {
  render() {
    return (
      <Provider store={this.props.store}>
        <RouterWithRedux>
            <Scene key="initializeLocalUser" hideNavBar component={InitializeLocalUser} title="Initialize" sharedResources={this.props.sharedResources}/>
            <Scene key="PWDashboard" component={PWDashboard} title="PWDashboard" sharedResources={this.props.sharedResources} />
        </RouterWithRedux>
      </Provider>
    );
  }
}
```

```
//PWDashboard.js

import React, { Component, PropTypes } from 'react';
import { View, ScrollView, Text } from 'react-native'
import TodoContainer from '../containers/TodoContainer';

export default class PWDashboard extends Component{
    constructor(props) {
        super(props)
    }
    render(){
        return(
            <View style={{flex: 1}}>
                <ScrollView>
                    <Text>my text</Text>
                    <TodoContainer sharedResources={this.props.sharedResources}  />
                </ScrollView>
                <View style={{position: 'absolute', left: 0, right: 0, bottom: 0}}><Text>My fixed footer</Text></View>
            </View>
        )
    }
}

```

I put in some logging and it's weird, I can see that dispatchToProps is setting the fetchTodosIfNeeded method as a prop in TodoContainer, so it's definitely doing something, but dispatch is not set as a prop. 

Also if I run connect() on the Dashboard component and then pass ITS dispatch prop manually down to TodoContainer it works, but that feels super hacky.

Here's my [package.json](https://gist.github.com/Prudwell/ec26fdbc31d3c2eb684c6a9ccc20f8f0)

Any ideas?

**Comment by markerikson** [comment]
This is working as expected.  If you do _not_ supply a `mapDispatch` function, `connect` will use its default.  That default `mapDispatch` simply gives you `this.props.dispatch`.  If you _do_ supply your own `mapDispatch` function, it's up to you what gets returned.  If you want to have some bound functions as props _and_ have `this.props.dispatch`, do so:

```js
const mapDispatch = (dispatch) => {
    const boundActionCreators = bindActionCreators(someActions, dispatch);
    const allActionProps = {...boundActionCreators, dispatch}
    return allActionProps;
}
```

**Comment by Prudwell** [comment]
Ah yes one of those times where I was doing it wrong all along and it only worked the previous way by luck!

My understanding was that **bindActionCreators(someActions, dispatch)** already did return the actions AND dispatch. I thought it was successfully providing dispatch to my TodoContainer when I rendered it as a Scene and couldn't figure out what changed when I loaded the component a different way. 

My guess is that it only worked because the \<RouterWithRedux\> component gets put through connect() and the Scene gets dispatch injected from it.

Thanks so much for you help!
